### PR TITLE
[OSD-22254] initial permissions for gcp wif

### PR DESCRIPTION
**Pull request by mrWinston**
### What type of PR is this?
_feature_

### What this PR does / why we need it?
This PR adds the first iteration of WIF templates. These can be seen as the gcp counterpart to the sts policies stored in this repo as well.

This PR must be merged before it's app-interface counterpart: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/105688

### Which Jira/Github issue(s) this PR fixes?

_Fixes [OSD-22254](https://issues.redhat.com//browse/OSD-22254)_

### CODE REVIEW

File: resources/wif/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+service accounts and permissions needed for the cluster's deployment and
+operation. This criteria is defined here, and then passed to OCM where it hosts
+an API endpoint for clients to query. The API provided by Clusters Service
+watches for changes in [this](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/data/services/ocm/shared-resources/common.yml)
+file. When this file changes a new [ConfigMap](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/resources/services/ocm/gcp-wif-template.configmap.yaml)
+is templated which Clusters Service pods (OCM) will read.
+
+## How to update WIF templates
+
+When Service Accounts, Roles and/or Permissions need to be updated, execute the following.
```
Comment (typeid) [inline resources/wif/README.md:14]: Can permissions be updated? AFAIU these permissions are consumed by the ocm-cli and changing them would also require customers to re-create the roles. I think for STS we don't update the permissions during a y stream.

Comment (fahlmant) [reply resources/wif/README.md:14]: That's the only time that STS permissions get updated on AWS clusters; as part of a y-stream upgrade.

Comment (typeid) [reply resources/wif/README.md:14]: Might have expressed it wrong but that's what I meant. The references piece of text sounds like upgrading within a y stream is a possibility.

Comment (mrWinston) [reply resources/wif/README.md]: I reworded the sentence slightly. Once I've updated the GAP analysis document for WIF, i will also link that here.

File: resources/wif/4.15/vanilla.yaml [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        predefined: true
+      - id: iam.ServiceAccountKeyAdmin
+        kind: Role
+        predefined: true
+      - id: iam.ServiceAccountUser
+        kind: Role
+        predefined: true
+      - id: storage.admin
+        kind: Role
+        predefined: true
```
Comment (typeid) [inline resources/wif/4.15/vanilla.yaml:32]: Shouldn't this one just need the installer permissions?

**Review by typeid — APPROVED** [review]
/lgtm
/approve

Let's merge this to unblock OCM and see if we missed anything in the permissions.

**Pull request merged**